src/Infrastructure/Services/TokenProviderService.cs

```diff
@@ -1,54 +1,53 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-using Microsoft.Extensions.Options;
-using Microsoft.IdentityModel.Clients.ActiveDirectory;
 using Contoso.FraudProtection.ApplicationCore.Interfaces;
 using Contoso.FraudProtection.Infrastructure.Utilities;
-using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+using Microsoft.Identity.Client;
 using System;
+using System.Threading.Tasks;
 
 namespace Contoso.FraudProtection.Infrastructure.Services
```

Comment:
The short version of the changes:
1. ADAL NuGet out, MSAL NuGet in.
1. The TokenProviderService now is a Singleton in the DI container.
1. The TokenProviderService builds a ConfidentialClientApplication during its constructor with either the secret or certificate details.
1. That client application is then used to get a token on demand.